### PR TITLE
feat: Add support for general-purpose function acceleration in Dynamo [6 / x]

### LINKED ISSUE #1959: ✨[Feature] Exclude key functions from tracing during AOT
# Context
As of #1921, we can exclude modules from acceleration in AOT using the global module-level registry. Still, there are certain functions which are also decomposed as a result of `aot_module_simplified`:
https://github.com/pytorch/TensorRT/blob/e10904941fe20743e96b21757bfb8b9bd4eda68c/py/torch_tensorrt/dynamo/backend/backends.py#L50-L55

These functions include `torch.functional.einsum`, which is decomposed as follows:
```python
##### THE MODEL
class SampleEinsum(torch.nn.Module):
    def forward(self, x):
        return torch.einsum('ii', x)

##### BEFORE AOT
graph():
    %l_x_ : torch.Tensor [#users=1] = placeholder[target=L_x_]
    %einsum : [#users=1] = call_function[target=torch.functional.einsum](args = (ii, %l_x_), kwargs = {})
    return (einsum,)

##### AFTER AOT
graph():
    %arg0_1 : [#users=1] = placeholder[target=arg0_1]
    %as_strided : [#users=1] = call_function[target=torch.ops.aten.as_strided.default](args = (%arg0_1, [4], [5], 0), kwargs = {})
    %permute : [#users=1] = call_function[target=torch.ops.aten.permute.default](args = (%as_strided, [0]), kwargs = {})
    %permute_1 : [#users=1] = call_function[target=torch.ops.aten.permute.default](args = (%permute, [0]), kwargs = {})
    %sum_1 : [#users=1] = call_function[target=torch.ops.aten.sum.dim_IntList](args = (%permute_1, [0]), kwargs = {})
    return (sum_1,)
```

# Feature Proposal
Augment the module-level replacement registry with a function-level replacement utility to replace TRT-convertible functions like `einsum` and exclude those from AOT tracing.

### PR DESCRIPTION
**Pull request by gs-olive**
# Description

- Add functionality for advanced exclusion of both function and module-type nodes in Torch-TRT
- Add sample exclusion for `torch.einsum` function which can be accelerated as a single unit via TRT
- Add utilities and improve module and function-level exclusion mechanisms
- Add test cases for new exclusion mechanism

Fixes #1959

## Type of change

- New feature (non-breaking change which adds functionality)

# Checklist:

- [ x ] My code follows the style guidelines of this project (You can use the linters)
- [ x ] I have performed a self-review of my own code
- [ x ] I have commented my code, particularly in hard-to-understand areas and hacks
- [ x ] I have made corresponding changes to the documentation
- [ x ] I have added tests to verify my fix or my feature
- [ x ] New and existing unit tests pass locally with my changes
- [ x ] I have added the relevant labels to my PR in so that relevant reviewers are notified

### CODE REVIEW

**Review by narendasan — APPROVED** [review]
LGTM
